### PR TITLE
Login/Signup pages refactor

### PR DESCRIPTION
**Pull request by AliSzu**
change form from `react hook form mui `to `react hook form`

### CODE REVIEW

File: src/types/FormTypes.ts [hunk 1]
```diff
@@ -10,14 +10,23 @@ export interface SignUpFormProps {
...
-  image?: FileList,
-  preparing: Preparing[],
+  title: string;
+  time: string;
+  description: string;
+  image?: FileList;
+  preparing: Preparing[];
   ingredients: Ingredient[];
-  imgSrc: string
+  imgSrc: string;
```
Comment (Szulerinio) [inline src/types/FormTypes.ts:29]: code formatter didn't catch that earlier? 🤔

File: src/components/atoms/PasswordField.tsx [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+    register,
+    formState: { errors },
+  } = useFormContext<PasswordFormValues>();
+
+  const handleClickShowPassword = () => setShowPassword((show) => !show);
+  const handleMouseDownPassword = (
+    event: React.MouseEvent<HTMLButtonElement>
+  ) => {
+    event.preventDefault();
+  };
```
Comment (Szulerinio) [inline src/components/atoms/PasswordField.tsx:39]: Why do you need this? 
What happens on mouse down? 🤔

**Review by Szulerinio — APPROVED** [review]

File: src/components/atoms/PasswordField.tsx (hunk 2 above)
Comment (AliSzu) [reply src/components/atoms/PasswordField.tsx:39]: onMouseDown allows the user to continue entering the password even when the password was shown/hidden by clicking the icon (thanks to the `event.preventDefault` in the handler). Without it, the field loses its focus, and the user has to manually click on the field again to start entering the password.

Comment (Szulerinio) [reply src/components/atoms/PasswordField.tsx:39]: Nice

File: src/types/FormTypes.ts (hunk 1 above)
Comment (AliSzu) [reply src/types/FormTypes.ts:29]: I think so??? For sure I didn't change it manually

Comment (Szulerinio) [reply src/types/FormTypes.ts:29]: That's what I hoped for 😅

**Pull request merged**